### PR TITLE
Fix issues with `clusterctl delete`

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:
Fixes issues with `clusterctl delete`
- Try to lookup machine instance using tags on delete
- Do not fail if VPC ID does not exist
- Update cluster-api dependency

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloud/aws/actuators/machine/actuator.go [hunk 1]
```diff
@@ -247,9 +247,15 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
...
-		// The machine hasn't been created yet
-		klog.Info("Instance is nil and therefore does not exist")
-		return nil
+		instance, err = ec2svc.InstanceByTags(scope)
+		if err != nil {
+			return errors.Errorf("failed to query instance by tags: %+v", err)
+		}
+		if instance == nil {
+			// The machine hasn't been created yet
+			klog.Info("Instance is nil and therefore does not exist")
```
Comment (chuckha) [inline pkg/cloud/aws/actuators/machine/actuator.go]: This should probably be behind a `.V(3)` or higher -- I'm trying to imagine a situation when I'm reading logs and I see this line pop up. I think it either needs more context or should not be in the Info log

File: pkg/cloud/aws/services/ec2/vpc.go [hunk 2]
```diff
@@ -130,10 +130,11 @@ func (s *Service) deleteVPC() error {
 	_, err := s.scope.EC2.DeleteVpc(input)
 	if err != nil {
 		// Ignore if it's already deleted
-		if code, ok := awserrors.Code(err); code != "InvalidVpcID.NotFound" && ok {
-			return errors.Wrapf(err, "failed to delete vpc %q", vpc.ID)
+		if code, ok := awserrors.Code(err); code == "InvalidVpcID.NotFound" && ok {
```
Comment (chuckha) [inline pkg/cloud/aws/services/ec2/vpc.go]: Nice, +1 on this change. the flow is much easier to read

File: pkg/cloud/aws/actuators/machine/actuator.go [hunk 3]
```diff
@@ -247,9 +247,15 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
...
 
 	if instance == nil {
-		// The machine hasn't been created yet
-		klog.Info("Instance is nil and therefore does not exist")
-		return nil
+		instance, err = ec2svc.InstanceByTags(scope)
+		if err != nil {
+			return errors.Errorf("failed to query instance by tags: %+v", err)
+		}
+		if instance == nil {
```
Comment (vincepri) [inline pkg/cloud/aws/actuators/machine/actuator.go]: This could be `else if`

File: pkg/cloud/aws/actuators/machine/actuator.go (hunk 1 above)
Comment (vincepri) [reply pkg/cloud/aws/actuators/machine/actuator.go]: +1

**Comment by detiber** [comment]
@chuckha @vincepri updated

**Pull request merged**